### PR TITLE
add sys variables `tidb_prefer_broadcast_join_by_exchange_data_size` to docs-cn

### PR DESCRIPTION
**Pull request by solotzg**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by zanmato1984 — APPROVED** [review]
LGTM

File: system-variables.md [hunk 1]
```diff
@@ -776,6 +776,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 单位：字节
 - 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
 
+### `tidb_prefer_broadcast_join_by_exchange_data_size` <span class="version-mark">从 v7.1 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- TiDB 选择 MPP Hash Join 算法时，会根据集群状态估算 Broadcast Hash Join 和 Shuffled Hash Join 所需进行网络交换的数据量，并优先选择网络交换数据量较小的算法。该功能开启后 [`tidb_broadcast_join_threshold_count`](#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 和 [`tidb_broadcast_join_threshold_size`](#tidb_broadcast_join_threshold_size-从-v50-版本开始引入) 将不再生效。
```
Comment (joey-yez) [inline system-variables.md:784]:
```suggestion
- 这个变量用于设定 TiDB 选择 MPP Hash Join 算法时，是否使用最小网络交换的数据量策略。使用该策略时，TiDB 会估算 Broadcast Hash Join 和 Shuffled Hash Join 两种算法所需进行网络交换的数据量，并选择网络交换数据量较小的算法。
- 该功能开启后 [`tidb_broadcast_join_threshold_count`](#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 和 [`tidb_broadcast_join_threshold_size`](#tidb_broadcast_join_threshold_size-从-v50-版本开始引入) 将不再生效。
```

File: system-variables.md [hunk 2]
```diff
@@ -776,6 +776,14 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 单位：字节
 - 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
 
+### `tidb_prefer_broadcast_join_by_exchange_data_size` <span class="version-mark">从 v7.1 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 这个变量用于设定 TiDB 选择 MPP Hash Join 算法时，是否使用最小网络交换的数据量策略。使用该策略时，TiDB 会估算 Broadcast Hash Join 和 Shuffled Hash Join 两种算法所需进行网络交换的数据量，并选择网络交换数据量较小的算法。
+- 该功能开启后 [`tidb_broadcast_join_threshold_count`](#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 和 [`tidb_broadcast_join_threshold_size`](#tidb_broadcast_join_threshold_size-从-v50-版本开始引入) 将不再生效。
```
Comment (qiancai) [inline system-variables.md:785]: 麻烦也请在 `tidb_broadcast_join_threshold_count` 和  `tidb_broadcast_join_threshold_size` 的变量描述补充下这个情况，谢谢

File: system-variables.md [hunk 3]
```diff
@@ -776,6 +776,14 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 单位：字节
 - 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
```
Comment (qiancai) [inline system-variables.md:778]: https://docs.pingcap.com/zh/tidb/dev/use-tiflash-mpp-mode#mpp-模式的算法支持 这篇文档里麻烦也请补充下这个 PR 里的信息
image

File: system-variables.md (hunk 2 above)
Comment (solotzg) [reply system-variables.md:787]: Done

File: system-variables.md (hunk 3 above)
Comment (solotzg) [reply system-variables.md:780]: Done

**Review by joey-yez — APPROVED** [review]

File: system-variables.md [hunk 4]
```diff
@@ -775,6 +776,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 开启 [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v71-版本开始引入) 功能后，该变量将不再生效。
+
+### `tidb_prefer_broadcast_join_by_exchange_data_size` <span class="version-mark">从 v7.1 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 这个变量用于设定 TiDB 选择 MPP Hash Join 算法时，是否使用最小网络交换的数据量策略。使用该策略时，TiDB 会估算 Broadcast Hash Join 和 Shuffled Hash Join 两种算法所需进行网络交换的数据量，并选择网络交换数据量较小的算法。
```
Comment (qiancai) [inline system-variables.md:786]:
```suggestion
- 这个变量用于设定 TiDB 选择 [MPP Hash Join 算法](/tiflash/use-tiflash-mpp-mode.md#mpp-模式的算法支持)时，是否使用最小网络交换的数据量策略。使用该策略时，TiDB 会估算 Broadcast Hash Join 和 Shuffled Hash Join 两种算法所需进行网络交换的数据量，并选择网络交换数据量较小的算法。
```

**Review by qiancai — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -3075,6 +3077,14 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
 
 </details>
 
+### `tidb_prefer_broadcast_join_by_exchange_data_size` <span class="version-mark">从 v7.1 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
```
Comment (solotzg) [inline system-variables.md:3084]:
```suggestion
- 默认值：`OFF`
```

File: system-variables.md [hunk 6]
```diff
@@ -775,6 +776,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 范围：`[0, 9223372036854775807]`
 - 单位：字节
 - 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- 开启 [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v71-版本开始引入) 功能后，该变量将不再生效。
+
+### `tidb_prefer_broadcast_join_by_exchange_data_size` <span class="version-mark">从 v7.1 版本开始引入</span>
```
[the remark this answers is not in the document]
Comment (solotzg) [reply system-variables.md]: done

File: system-variables.md [hunk 7]
```diff
@@ -766,6 +766,7 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 范围：`[0, 9223372036854775807]`
 - 单位：行
 - 如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- 开启 [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v71-版本开始引入) 功能后，该变量将不再生效。
```
Comment (ran-huang) [inline system-variables.md:769]:
```suggestion
- 开启 [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v710-版本开始引入) 功能后，该变量将不再生效。
```

File: system-variables.md [hunk 8]
```diff
@@ -775,6 +776,7 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 范围：`[0, 9223372036854775807]`
 - 单位：字节
 - 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- 开启 [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v71-版本开始引入) 功能后，该变量将不再生效。
```
Comment (ran-huang) [inline system-variables.md:779]:
```suggestion
- 开启 [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v710-版本开始引入) 功能后，该变量将不再生效。
```

File: system-variables.md [hunk 9]
```diff
@@ -3075,6 +3077,14 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
 
 </details>
 
+### `tidb_prefer_broadcast_join_by_exchange_data_size` <span class="version-mark">从 v7.1 版本开始引入</span>
```
Comment (ran-huang) [inline system-variables.md:3080]:
```suggestion
### `tidb_prefer_broadcast_join_by_exchange_data_size` <span class="version-mark">从 v7.1.0 版本开始引入</span>
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: tiflash/use-tiflash-mpp-mode.md [hunk 10]
```diff
@@ -95,10 +95,11 @@ mysql> explain select count(*) from customer c join nation n on c.c_nationkey=n.
...
 在执行计划中，出现了 `ExchangeReceiver` 和 `ExchangeSender` 算子。该执行计划表示 `nation` 表读取完毕后，经过 `ExchangeSender` 算子广播到各个节点中，与 `customer` 表先后进行 `HashJoin` 和 `HashAgg` 操作，再将结果返回至 TiDB 中。
 
-TiFlash 提供了两个全局/会话变量决定是否选择 Broadcast Hash Join，分别为：
+TiFlash 提供了 3 个全局/会话变量决定是否选择 Broadcast Hash Join，分别为：
 
 - [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为 bytes。如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
 - [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v71-版本开始引入)，控制是否使用最小网络数据交换策略。使用该策略时，TiDB 会估算 Broadcast Hash Join 和 Shuffled Hash Join 两种算法所需进行网络交换的数据量，并选择网络交换数据量较小的算法。该功能开启后， [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 和 [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 将不再生效。
```
Comment (solotzg) [inline tiflash/use-tiflash-mpp-mode.md:102]:
```suggestion
- [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v710-版本开始引入)，控制是否使用最小网络数据交换策略。使用该策略时，TiDB 会估算 Broadcast Hash Join 和 Shuffled Hash Join 两种算法所需进行网络交换的数据量，并选择网络交换数据量较小的算法。该功能开启后， [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 和 [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 将不再生效。
```

File: tiflash/use-tiflash-mpp-mode.md [hunk 11]
```diff
@@ -95,10 +95,11 @@ mysql> explain select count(*) from customer c join nation n on c.c_nationkey=n.
 
 在执行计划中，出现了 `ExchangeReceiver` 和 `ExchangeSender` 算子。该执行计划表示 `nation` 表读取完毕后，经过 `ExchangeSender` 算子广播到各个节点中，与 `customer` 表先后进行 `HashJoin` 和 `HashAgg` 操作，再将结果返回至 TiDB 中。
 
-TiFlash 提供了两个全局/会话变量决定是否选择 Broadcast Hash Join，分别为：
+TiFlash 提供了 3 个全局/会话变量决定是否选择 Broadcast Hash Join，分别为：
 
 - [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为 bytes。如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
```
Comment (solotzg) [inline tiflash/use-tiflash-mpp-mode.md:100]:
```suggestion
- [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_size-从-v50-版本开始引入)，单位为 bytes。如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
```

File: tiflash/use-tiflash-mpp-mode.md [hunk 12]
```diff
@@ -95,10 +95,11 @@ mysql> explain select count(*) from customer c join nation n on c.c_nationkey=n.
...
 - [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v710-版本开始引入)，控制是否使用最小网络数据交换策略。使用该策略时，TiDB 会估算 Broadcast Hash Join 和 Shuffled Hash Join 两种算法所需进行网络交换的数据量，并选择网络交换数据量较小的算法。该功能开启后， [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 和 [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 将不再生效。
```
Comment (solotzg) [inline tiflash/use-tiflash-mpp-mode.md:102]:
```suggestion
- [`tidb_prefer_broadcast_join_by_exchange_data_size`](/system-variables.md#tidb_prefer_broadcast_join_by_exchange_data_size-从-v710-版本开始引入)，控制是否使用最小网络数据交换策略。使用该策略时，TiDB 会估算 Broadcast Hash Join 和 Shuffled Hash Join 两种算法所需进行网络交换的数据量，并选择网络交换数据量较小的算法。该功能开启后， [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_size-从-v50-版本开始引入) 和 [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入) 将不再生效。
```

**Pull request merged**